api/src/pcapi/core/finance/models.py

```diff
@@ -983,10 +983,18 @@ class FinanceIncident(PcObject, Base, Model):
     def is_partial(self) -> bool:
         return any(booking_incident.is_partial for booking_incident in self.booking_finance_incidents)
 
-    @property
+    @hybrid_property
     def relates_to_collective_bookings(self) -> bool:
         return any(booking_incident.collectiveBooking for booking_incident in self.booking_finance_incidents)
 
+    @relates_to_collective_bookings.expression  # type: ignore[no-redef]
+    def relates_to_collective_bookings(cls) -> sqla.sql.elements.UnaryExpression:  # pylint: disable=no-self-argument
+        aliased_booking_finance_incident = sqla.orm.aliased(BookingFinanceIncident)
+        return sqla.exists().where(
+            aliased_booking_finance_incident.incidentId == cls.id,
+            aliased_booking_finance_incident.collectiveBooking != None,
```

Comment:
Filtre sur la colonne `collectiveBookingId` plutôt que l'objet de l'ORM `collectiveBooking`.
Aussi, un détail : Utilise l'opérateur `is_not` pour adopter nos conventions de syntaxe, afin d'avoir un code homogène.
```suggestion
            aliased_booking_finance_incident.collectiveBookingId.is_not(None),
```